File: python/2003.py
while True:
    try:
        hour, minutes = input().split(':')
        hour = int(hour)
        minutes = int(minutes)

        minutes += 60
        lateH = 0
        lateM = 0

        if minutes >= 60:
            hour += 1
            minutes -= 60

        if hour >= 8:
            lateH = hour - 8
            lateM = minutes

        late = (lateH * 60) + lateM

        print(f'Atraso maximo: {late}')
    except:
        break

File: lua/2003.lua
local function main()
    local time = io.read()
    local hour = tonumber(time:sub(1,1))
    local minutes = tonumber(time:sub(3,4))
    minutes = (minutes + 60) % 60
    hour = hour + 1
    local delay = 0
    if hour >= 8 then
        delay = (hour - 8) * 60 + minutes
    end
    print('Atraso maximo: '..tostring(delay))
end

while true do
    if not pcall(main) then
        break
    end
end